File: FactorialOfNumberRecursiveApproach/Fact.java
/*
    Author : Pakshal Shashikant Jain 
    Date : 08/05/2024
    Program : Write a recursive program which accept number from user and return its
              factorial.
              
              Input : 5
              Output : 120 
*/

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Required Libraries
import java.lang.*;
import java.util.*;

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

class Factorial
{
    //Characteristics of Class
    int ino;
    int imult;

    //Constructor of Class
    Factorial(int value)
    {
        this.ino = value;
        this.imult = 1;
    }

    //Behaviour of Class To Calculate Factorial
    int NumFact()
    {
        if(this.ino != 0)
        {
            this.imult = this.imult * this.ino;
            this.ino--;
            this.NumFact();
        }

        return this.imult;
    }
}

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Class
class Fact 
{
    //Main Function
    public static void main(String args[])
    {
        int ino = 0;
        int iret = 0;

        //Creating object of Scanner Class To Take Input From User
        Scanner sobj = new Scanner(System.in);

        System.out.println("Jay Ganesh.....");

        //Taking Input From user
        System.out.println("Enter One Number : ");
        ino = sobj.nextInt();

        //Creating Object of Factoriial Class
        Factorial fobj = new Factorial(ino);
        
        //Call To Behaviour of Class
        iret = fobj.NumFact();

        //Printing Factorial Calculated
        System.out.printf("Factorial of Number is : %d",iret);
    }
}

File: FactorialOfNumberRecursiveApproach/Fact.c
/*
    Author : Pakshal Shashikant Jain 
    Date : 08/05/2024
    Program : Write a recursive program which accept number from user and return its
              factorial.
              
              Input : 5
              Output : 120 
*/

//Required Libraries
#include<stdio.h>

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Function To Calculate Factorial
int Fact(int iNo)
{
    static int imult = 1;

    if(iNo != 0)
    {
        imult = imult * iNo;
        iNo--;

        //Recursive Function Call To Fact Function
        Fact(iNo);
    }
    else 
    {
        return imult;
    }
}

//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Function 
int main()
{
    int ino = 0;
    int iret = 0;

    printf("Jay Ganesh.....\n");
    
    //Taking Input From User
    printf("Enter One Number : ");
    scanf("%d",&ino);

    //Call To Factorial Function 
    iret = Fact(ino);

    //Printing Factorial Calculated
    printf("Factorial of %d is : %d",ino,iret);

    return 0;
}